corehq/apps/app_manager/models.py

```diff
@@ -986,9 +986,80 @@ def validate(self, value, required=True):
 
 
 class NavMenuItemMediaMixin(DocumentSchema):
+    """
+        Language-specific icon and audio.
+        Properties are map of lang-code to filepath
+    """
+    media_image = SchemaDictProperty(JRResourceProperty)
+    media_audio = SchemaDictProperty(JRResourceProperty)
+
+    @classmethod
+    def wrap(cls, data):
+        # ToDo - Remove after migration
+        for media_attr in ('media_image', 'media_audio'):
+            old_media = data.get(media_attr, None)
+            if old_media and isinstance(old_media, basestring):
+                new_media = {'default': old_media}
+                data[media_attr] = new_media
+
+        return super(NavMenuItemMediaMixin, cls).wrap(data)
+
+    def _get_media_by_language(self, media_attr, lang):
+        assert media_attr in ('media_image', 'media_audio')
+
+        media_dict = getattr(self, media_attr)
+        if not media_dict:
+            return None
+        if lang in media_dict:
+            return media_dict[lang]
+        else:
+            # if the queried lang key doesn't exist,
+            # return the first in the sorted list
+            for lang, item in sorted(media_dict.items()):
+                return item
+
+    @property
+    def default_media_image(self):
+        return self.icon_by_language('en')
+
+    @property
+    def default_media_audio(self):
+        return self.audio_by_language('en')
```

Comment:
A comment for these methods would be good explaining that they are for legacy apps.